### PR TITLE
Convert ConflictMarkerResolutionTests to the new test library

### PR DESCRIPTION
**Pull request by sharwell**

### CODE REVIEW

File: src/EditorFeatures/VisualBasicTest/ConflictMarkerResolution/ConflictMarkerResolutionTests.vb [hunk 1]
```diff
@@ -238,7 +278,16 @@ namespace N
...
     class Program4
     end class
-end namespace", index:=2)
+end namespace"
+
+            Await New VerifyVB.Test With
+            {
+                .TestCode = source,
+                .FixedCode = fixedSource,
+                .NumberOfIncrementalIterations = 2,
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/VisualBasicTest/ConflictMarkerResolution/ConflictMarkerResolutionTests.vb]: 1. What is `NumberOfIncrementalIterations` for?
2. Why does the test pass in `.CodeActionEquivalenceKey`.  That seems wrong.  The test shouldn't  have to know what equivalence key was used.  That is strictly an internal detail.

Comment (sharwell) [reply src/EditorFeatures/VisualBasicTest/ConflictMarkerResolution/ConflictMarkerResolutionTests.vb]: > What is `NumberOfIncrementalIterations` for?

By default, the test verifies that the total number of incremental fixes applied to reach the final state is less than or equal to the original number of fixable diagnostics, which for this test is 6. I set the value to 2 to explicitly verify that it takes exactly 2 iterations to reach the final state with no remaining diagnostics.

> Why does the test pass in `.CodeActionEquivalenceKey`.

By default, the first fix is used. If you pass the index or equivalence key, the matching one is used. If you pass both, which occurs here, the test validates that the code action at the specified index is same as the code action with the specified equivalence key, which allows us to validate the positioning of known fixes within the list.

The preferable approach for readable tests is to either pass the equivalence key or pass both.

Comment (CyrusNajmabadi) [reply src/EditorFeatures/VisualBasicTest/ConflictMarkerResolution/ConflictMarkerResolutionTests.vb]: ok.  so the CodeActionEquivalenceKey is just for vlaidation purposes.  it doesn't change teh actual code that the test runs or how it calls into the underlying system.  and, if it was gone, we'd still get the same results.  Is that correct?

If that's the case, I'm ok with this :)

Comment (sharwell) [reply src/EditorFeatures/VisualBasicTest/ConflictMarkerResolution/ConflictMarkerResolutionTests.vb]: Yes, that is correct.

**Pull request merged**